### PR TITLE
More explicitly differentiate between iterator helper methods/objects

### LINKED ISSUE #36699: Ambiguity of concept "iterator helper"
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/map

### What specific section or headline is this issue about?

Return value

### What information was incorrect, unhelpful, or incomplete?

Throughout the article the term "iterator helper" is used for the returned iterator

### What did you expect to see?

The `map` method returns an iterator, not iterator helper.

### Do you have any supporting links, references, or citations?

The page hyperlinks "iterator helper" to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator#iterator_helpers, which defines "iterator helpers" as the *set of methods* on iterators, not the iterator itself. This is inconsistent.

### Do you have anything more you want to share?

A similar problem occurs on other pages on iterator helpers, like `filter`, `drop`, `take`,...etc.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/iterator/map`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/map
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/iterator/map/index.md
* Last commit: https://github.com/mdn/content/commit/06b418a190b8e4a46682ab706d14984e7db34862
* Document last modified: 2024-09-11T03:36:25.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
Fix https://github.com/mdn/content/issues/36699. @hamishwillee has expressed the same concern before.

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/iterator/index.md [hunk 1]
```diff
@@ -90,7 +90,11 @@ You will find many iterator methods analogous to array methods, such as:
 
 {{jsxref("Iterator.prototype.drop()")}} and {{jsxref("Iterator.prototype.take()")}} combined are somewhat analogous to {{jsxref("Array.prototype.slice()")}}.
 
-Among these methods, {{jsxref("Iterator/filter", "filter()")}}, {{jsxref("Iterator/flatMap", "flatMap()")}}, {{jsxref("Iterator/map", "map()")}}, {{jsxref("Iterator/drop", "drop()")}}, and {{jsxref("Iterator/take", "take()")}} return a new _Iterator Helper_ object. The iterator helper is also an `Iterator` instance, making the helper methods chainable. All iterator helper objects inherit from a common prototype object, which implements the iterator protocol:
+### Iterator helper objects
+
+> [!NOTE] > _Iterator helper objects_ and _iterator helper methods_ are two different concepts. An Iterator helper object is detectable at runtime, while "iterator helper method" is just a name for a set of methods for comprehension. _Iterator helper_ may refer to either the object or the method, depending on the context.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/iterator/index.md:95]:
```suggestion
> **Note:** _Iterator helper objects_ and _iterator helper methods_ are two different concepts. An Iterator helper object is detectable at runtime, while "iterator helper method" is just a name for a set of methods for comprehension. _Iterator helper_ may refer to either the object or the method, depending on the context.
```

File: files/en-us/web/javascript/reference/global_objects/iterator/index.md [hunk 2]
```diff
@@ -90,7 +90,11 @@ You will find many iterator methods analogous to array methods, such as:
 
 {{jsxref("Iterator.prototype.drop()")}} and {{jsxref("Iterator.prototype.take()")}} combined are somewhat analogous to {{jsxref("Array.prototype.slice()")}}.
 
-Among these methods, {{jsxref("Iterator/filter", "filter()")}}, {{jsxref("Iterator/flatMap", "flatMap()")}}, {{jsxref("Iterator/map", "map()")}}, {{jsxref("Iterator/drop", "drop()")}}, and {{jsxref("Iterator/take", "take()")}} return a new _Iterator Helper_ object. The iterator helper is also an `Iterator` instance, making the helper methods chainable. All iterator helper objects inherit from a common prototype object, which implements the iterator protocol:
+### Iterator helper objects
+
+> **Note:** _Iterator helper objects_ and _iterator helper methods_ are two different concepts. An Iterator helper object is detectable at runtime, while "iterator helper method" is just a name for a set of methods for comprehension. _Iterator helper_ may refer to either the object or the method, depending on the context.
```
Comment (pepelsbey) [inline files/en-us/web/javascript/reference/global_objects/iterator/index.md:95]:
```suggestion
> [!NOTE]
> _Iterator helper objects_ and _iterator helper methods_ are two different concepts. An Iterator helper object is detectable at runtime, while "iterator helper method" is just a name for a set of methods for comprehension. _Iterator helper_ may refer to either the object or the method, depending on the context.
```

Let’s use [the latest note syntax](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#notes_warnings_and_callouts).

**Review by pepelsbey — APPROVED** [review]
Looks good! Just a small syntax suggestion. Thank you :)

File: files/en-us/web/javascript/reference/global_objects/iterator/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/iterator/index.md:95]: See the note in the section you linked:

> Currently, due to a Prettier bug, the GFM alert syntax cannot be used if the first character of a note or warning is a formatting symbol, such as a backquote, asterisk, square bracket or curly bracket. In this case, use the old syntax > **Note:** instead. Writers are not required to rephrase the content to work around the formatter.

**Pull request merged**